scripts/run-pyo3-greptime.sh

```diff
@@ -0,0 +1,155 @@
+
+# This script should be compatible both in Linux and macOS
+OS_TYPE="$(uname)"
+readonly OS_TYPE
+
+check_command_existence() {
+    command -v "$1" &> /dev/null
+}
+
+get_python_version() {
+    case "$OS_TYPE" in
+        Darwin)
+            otool -L $GREPTIME_EXEC_PATH | grep -o 'Python.framework/Versions/3.[0-9]\+/Python' | grep -o '3.[0-9]\+'
+            ;;
+        Linux)
+            ldd $GREPTIME_EXEC_PATH | grep -o 'libpython3\.[0-9]\+' | grep -o '3\.[0-9]\+'
+            ;;
+        *)
+            echo "Unsupported OS type: $OS_TYPE"
+            exit 1
+            ;;
+    esac
+}
+
+setup_virtualenv() {
+    local req_py_version="$1"
+    local env_name="GreptimeTmpVenv$req_py_version"
+    virtualenv --python=python"$req_py_version" "$env_name"
+    source "$env_name/bin/activate"
+}
+
+setup_conda_env() {
+    local req_py_version="$1"
+    local conda_base
+    conda_base=$(conda info --base) || { echo "Error obtaining conda base directory"; exit 1; }
+    . "$conda_base/etc/profile.d/conda.sh"
+
+    if ! conda list --name "GreptimeTmpPyO3Env$req_py_version" &> /dev/null; then
+        conda create --yes --name "GreptimeTmpPyO3Env$req_py_version" python="$req_py_version"
+    fi
+
+    conda activate "GreptimeTmpPyO3Env$req_py_version"
+}
+
+get_optional_args(){
+    # if not set by --path path-of-greptime-executable
+    # default to search local folder for greptime executable
+    ARGS=$(getopt -o "p:y::e:" -l "path:,yes::,env:" -- "$@")
+    # assign ARGS to positional parameters $1, $2, etc...
+    eval set -- "$ARGS"
+    unset ARGS
+    # default path to executable
+    exec_path="./greptime"
+    # parse for path
+    while true; do
+        case "$1" in
+            -p|--path)
+                shift
+                exec_path="$1"
+                shift
+                ;;
+            -y|--yes)
+                shift
+                export YESMAN=1
+                shift
+                ;;
+            -e|--env)
+                shift
+                export USE_ENV="$1"
+                shift
+                ;;
+            --)
+                shift
+                break
+                ;;
+            *)
+                echo "Error parsing arguments: $1"
+                break
+                ;;
+        esac
+    done
+    export GREPTIME_EXEC_PATH=$exec_path
+    export REST_OF_ARGS=$@
+}
+
+# Set library path and pass all arguments to greptime to run it
+execute_greptime() {
+    if [[ "$OS_TYPE" == "Darwin" ]]; then
+        DYLD_LIBRARY_PATH="${CONDA_PREFIX:-$PREFIX}/lib:${LD_LIBRARY_PATH:-}" $GREPTIME_EXEC_PATH $@
+    elif [[ "$OS_TYPE" == "Linux" ]]; then
+        LD_LIBRARY_PATH="${CONDA_PREFIX:-$PREFIX}/lib:${LD_LIBRARY_PATH:-}" $GREPTIME_EXEC_PATH $@
+    fi
+}
+
+main() {
+    get_optional_args $@
+    echo Path of greptime executable: $GREPTIME_EXEC_PATH
+    echo Args passed to greptime executable: $REST_OF_ARGS
```

Comment:
The print logic can be aggregated in `get_optional_args`(or `parse_args()` will be better).